paasta_tools/cli/cmds/validate.py

```diff
@@ -587,10 +694,9 @@ def validate_autoscaling_configs(service_path):
         for instance, instance_config in load_all_instance_configs_for_service(
             service=service, cluster=cluster, soa_dir=soa_dir
         ):
-
             if (
                 instance_config.get_instance_type() in K8S_TYPES
-                and instance_config.is_autoscaling_enabled()
+                and instance_config.is_autoscaling_enabled()  # type: ignore
```

Comment:
just curious: what was the error here? i assume something to do with the weird way the instance config classes are setup inheritance-wise?

Reply:
`instance_config` is a generic `InstanceConfig` object, which has no `is_autoscaling_enabled` method.  The check above `get_instance_type() in K8S_TYPES` narrows it down to a KubernetesDeploymentConfig which does have an `is_autoscaling_enabled` method.  The alternate way to do this would be to assert that instance_config is either a `KubernetesDeploymentConfig` or an `EksDeploymentConfig` but..... that seems ugly.

I'll add a comment here.

Reply:
Actually -- I can do a cast to LongRunningServiceConfig, which is slightly more explicit.